AEPMedia/Sources/MediaCollection/MediaRealTimeSession.swift

```diff
@@ -178,18 +179,59 @@ class MediaRealTimeSession: MediaSession {
+            self?.trySendHit()
+        }
+    }
+
+    private func shouldRetry(eventType: String, connection: HttpConnection?) -> Bool {
+        guard eventType == MediaConstants.MediaCollection.EventType.SESSION_START else {
+            Log.trace(label: Self.LOG_TAG, "[\(Self.CLASS_NAME)<\(#function)>] - [Session (\(self.id))] \(eventType) request resulted in error and since event type is not sessionStart it will be dropped.")
+            return false
+        }
+
+        guard sessionStartRetryCount < MediaRealTimeSession.MAX_ALLOWED_FAILURE else {
+            Log.trace(label: Self.LOG_TAG, "[\(Self.CLASS_NAME)<\(#function)>] - [Session (\(self.id))] \(eventType) request resulted in error but since it surpassed the retry limit:(\(MediaRealTimeSession.MAX_ALLOWED_FAILURE), the request will be dropped.")
+            return false
+        }
+
+        guard let connection = connection else {
+            // should reach here when no connection (http, url) error
+            // retry for case when error occured while parsing sessionStart response for sessionId
+            Log.trace(label: Self.LOG_TAG, "[\(Self.CLASS_NAME)<\(#function)>] - [Session (\(self.id))] Unable to extract sessionId from sessionStart request response, the request will be retried.")
+            return true
+        }
+
+        if let urlError = connection.error as? URLError {
+
+            if urlError.isRecoverable {
+                Log.debug(label: Self.LOG_TAG, "[\(Self.CLASS_NAME)<\(#function)>] - [Session(\(self.id))] \(eventType) request failed with recoverable URL error:(\(urlError.localizedDescription)) code:(\(urlError.errorCode)). Request will be retried.")
+                return true // retry for recoverable url errors
+            } else {
+                Log.debug(label: Self.LOG_TAG, "[\(Self.CLASS_NAME)<\(#function)>] - [Session(\(self.id))] \(eventType) request failed with unrecoverable URL error:(\(urlError.localizedDescription)) code:(\(urlError.errorCode)). Request will be dropped.")
+                return false
+            }
+        }
+
+        if let responseCode = connection.responseCode {
+
+            if NetworkServiceConstants.RECOVERABLE_ERROR_CODES.contains(responseCode) {
+                Log.debug(label: Self.LOG_TAG, "[\(Self.CLASS_NAME)<\(#function)>] - [Session(\(self.id))] \(eventType) request failed with recoverable HTTP error:(\(connection.responseMessage ?? "")) response code:(\(responseCode)). Request will be retried.")
+                return true // retry for recoverable http errors
+            } else {
+                Log.debug(label: Self.LOG_TAG, "[\(Self.CLASS_NAME)<\(#function)>] - [Session(\(self.id))] \(eventType) request failed with unrecoverable HTTP error:(\(connection.responseMessage ?? "")) response code:(\(connection.responseCode ?? -1)). Request will be dropped.")
+                return false
             }
         }
+
+        return false
```

Comment:
Add a trace log here. I'm assuming we shouldn't arrive here, but log out here that the sessionStart response has a connection without an error or responseCode, so we are dropping the request.